pkg/skaffold/parser/config.go

```diff
@@ -179,7 +179,7 @@ func processEachConfig(config *latest_v1.SkaffoldConfig, cfgOpts configOpts, opt
 }
 
 // filterActiveProfiles selects the set of profiles to activate in the dependency config based on the current set of active profiles.
-func filterActiveProfiles(d latest_v1.ConfigDependency, profiles []string) []string {
+func filterActiveProfiles(d latest_v1.ConfigDependency, profiles []string, opts config.SkaffoldOptions) []string {
 	var depProfiles []string
 	for _, ap := range d.ActiveProfiles {
```

Comment:
We should be including the propagated profiles here too, right?  (My suggestion above will do this.)